test/e2e/kubectl/logs.go

```diff
@@ -33,13 +38,55 @@ import (
 	e2ekubectl "k8s.io/kubernetes/test/e2e/framework/kubectl"
 	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
 	e2eoutput "k8s.io/kubernetes/test/e2e/framework/pod/output"
+	e2estatefulset "k8s.io/kubernetes/test/e2e/framework/statefulset"
 	imageutils "k8s.io/kubernetes/test/utils/image"
 	admissionapi "k8s.io/pod-security-admission/api"
-
-	"github.com/onsi/ginkgo/v2"
-	"github.com/onsi/gomega"
 )
 
+func testingStatefulSet(name, ns string, numberOfPods int32) appsv1.StatefulSet {
+	return appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: ns,
+			Labels: map[string]string{
+				"name": name,
+			},
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Replicas: &numberOfPods,
+			Selector: &metav1.LabelSelector{
+				MatchLabels: map[string]string{
+					"name": name,
+				},
+			},
+			Template: v1.PodTemplateSpec{
+				ObjectMeta: metav1.ObjectMeta{
+					Labels: map[string]string{
+						"name": name,
+					},
+					Annotations: map[string]string{
+						podcmd.DefaultContainerAnnotationName: "container-2",
+					},
+				},
+				Spec: v1.PodSpec{
+					Containers: []v1.Container{
+						{
+							Name:  "container-1",
+							Image: imageutils.GetE2EImage(imageutils.Agnhost),
+							Args:  []string{"logs-generator", "--log-lines-total", "10", "--run-duration", "5s"},
+						},
+						{
+							Name:  "container-2",
+							Image: imageutils.GetE2EImage(imageutils.Agnhost),
+							Args:  []string{"logs-generator", "--log-lines-total", "20", "--run-duration", "5s"},
+						},
+					},
+					RestartPolicy: v1.RestartPolicyAlways,
```

Comment:
This might be the actual cause of the failures. Let's try setting this to never. What might be happening is that the pods are execting before they can be succesfully recongized as started by the wait.

Reply:
Unfortunately seems as though that is invalid

```bash
 [FAILED] Failed to create StatefulSet: StatefulSet.apps "sts7c9438e1-d736-4c2a-ae3e-cbb9ef945d32" is invalid: spec.template.spec.restartPolicy: Unsupported value: "Never": supported values: "Always"
  In [BeforeEach] at: k8s.io/kubernetes/test/e2e/kubectl/logs.go:274
```